lib/env.h

```diff
@@ -49,6 +49,8 @@
 #  define BASE64_WORDSIZE __WORDSIZE
 #elif defined (__LONG_WIDTH__)
 #  define BASE64_WORDSIZE __LONG_WIDTH__
+#elif defined(__SIZEOF_LONG__) && defined(__CHAR_BIT__) && (__CHAR_BIT__ == 8)
+#  define BASE64_WORDSIZE (__SIZEOF_LONG__ * 8)
```

Comment:
> I think we can assume that __CHAR_BIT__ == 8

If we assume, let's just enforce that the assumption is correct no ?

I amende the commit message.

> Why  `sizeof(long) * 8`

That's 64 on 64-bits arch and 32 on 32-bits arch, same as `__LONG_WIDTH__ ` just above but not defined with clang.

Reply:
I think `__SIZE_WIDTH__` or `__POINTER_WIDTH__` might be more applicable, because `size_t` and pointers are pretty much always expressed in the native word size.

Reply:
> I think __SIZE_WIDTH__ or __POINTER_WIDTH__ might be more applicable, because size_t and pointers are pretty much always expressed in the native word size.

not on x86_32

Reply:
I don't like that `long` stuff either but that's what makes the most sense to me right now...

Reply:
Alright, fair enough. The only place where `BASE64_WORDSIZE` is used is to choose between 32-bit and 64-bit generic codecs, so nothing should break if we accidentally pick the wrong one.

Reply:
I'm reworking this a bit differently with `__SIZE_WIDTH__` & proper detection for x32 ABI.